packages/create-react-native-library/templates/native-common/android/build.gradle

```diff
@@ -112,69 +152,24 @@ repositories {
   mavenCentral()
   google()
 
-  def found = false
-  def defaultDir = null
-  def androidSourcesName = 'React Native sources'
-
-  if (rootProject.ext.has('reactNativeAndroidRoot')) {
-    defaultDir = rootProject.ext.get('reactNativeAndroidRoot')
-  } else {
-    defaultDir = new File(
-      projectDir,
-      '/../../../node_modules/react-native/android'
-    )
-  }
+  if (REACT_NATIVE_MAJOR_VERSION == 0 && REACT_NATIVE_MINOR_VERSION < 71) {
+    def androidSourcesDir = null
+    def androidSourcesName = 'React Native sources'
 
-  if (defaultDir.exists()) {
-    maven {
-      url defaultDir.toString()
-      name androidSourcesName
+    if (rootProject.ext.has('reactNativeAndroidRoot')) {
+      androidSourcesDir = rootProject.ext.get('reactNativeAndroidRoot')
+    } else {
+      androidSourcesDir = new File(reactNativeRootDir, '/android')
     }
 
-    logger.info(":${project.name}:reactNativeAndroidRoot ${defaultDir.canonicalPath}")
-    found = true
-  } else {
-    def parentDir = rootProject.projectDir
-
-    1.upto(5, {
-      if (found) return true
-      parentDir = parentDir.parentFile
-
-      def androidSourcesDir = new File(
-        parentDir,
-        'node_modules/react-native'
-      )
-
-      def androidPrebuiltBinaryDir = new File(
-        parentDir,
-        'node_modules/react-native/android'
```

Comment:
I'm looking into re-creating the `react-native/android/` folder just to short-circut this logic.

But yes it should be cleaned up.

Reply:
@cortinico at least for code generated by bob, the absence of this folder wouldn't cause an issue. previously it was happy as long as it found the `react-native` folder in `node_modules` - though the original code was from `react-native-webview` repo so I don't have the context of why.